test/Winton.Extensions.Configuration.Consul.Test/Extensions/KVPairQueryResultExtensionsTests.cs

```diff
@@ -165,10 +165,15 @@ public sealed class HasValue : KVPairQueryResultExtensionsTests
         public class ToConfigDictionary : KVPairQueryResultExtensionsTests
         {
             private readonly Mock<IConfigurationParser> _parser;
+            private readonly IConsulConfigurationSource _source;
```

Comment:
I think we can actually make some nice improvements to these tests now. Previously, the implementation of `ToConfigDictionary` was coupled to the extension method `KVPair.ConvertToConfig`. With these changes that's no longer the case. That means that we can better mock the behaviour in this test suite. Rather than having to construct an `IConfigurationParser` and an `IConsulConfigurationSource` we can just pass a made up `Func` into each test invocation of `result.ToConfigDictionary`. I'll give a specific example below.